.gitlab-ci.yml

```diff
@@ -72,22 +84,54 @@ workflow:
 .start-substrate-contracts-node-rand-extension:      &start-substrate-contracts-node-rand-extension
   - substrate-contracts-node-rand-extension --tmp --dev -lruntime::contracts=debug > /tmp/substrate-contracts-node-rand-extension.log 2>&1 &
 
+.shutdown-substrate-contracts-node:                  &shutdown-substrate-contracts-node
+  - pkill -f "substrate-contracts-node --tmp --dev"
+
+.shutdown-substrate-contracts-node-rand-extension:   &shutdown-substrate-contracts-node-rand-extension
+  - pkill -f "substrate-contracts-node-rand-extension --tmp --dev"
+
+
+# needed vars have to be "exported" on a earlier stage
+parent-vars:
+  stage:                                             declare-vars
+  <<:                                                *kubernetes-env
+  script:
+    - BRANCH="master"
+    - REPO="https://github.com/paritytech/ink.git"
+    - REPO_NAME="ink"
+    - echo ${TRGR_REF}
+    - if [ -n "$TRGR_REF" ] && [ "$TRGR_REF" != "master" ]; then
+        PR_JSON=`curl -s https://api.github.com/repos/paritytech/ink/pulls/${TRGR_REF}`;
+        BRANCH=`echo "${PR_JSON}" | jq -r .head.ref`;
+        REPO=`echo "${PR_JSON}" | jq -r .head.repo.git_url`;
+        REPO_NAME=`echo "${PR_JSON}" | jq -r .head.repo.name`;
+      fi
+    - echo "BRANCH=${BRANCH}" | tee -a parent-vars.env
+    - echo "REPO=${REPO}" | tee -a parent-vars.env
+    - echo "REPO_NAME=${REPO_NAME}" | tee -a parent-vars.env
+    - echo "REDIS_SIZES_KEY=${CI_PROJECT_NAME}::${REPO_NAME}::${BRANCH}::sizes" | tee -a parent-vars.env
+    - echo "REDIS_SIZES_KEY_MASTER=${CI_PROJECT_NAME}::results::${REPO_NAME}::sizes" | tee -a parent-vars.env
+  artifacts:
+    reports:
+      dotenv: parent-vars.env
+
+
 .build-contracts:                                    &build-contracts
+  # delete old list items if the key has existed previously
+  - redis-cli -u $GITLAB_REDIS_URI del $REDIS_SIZES_KEY
+  - echo "Data will be written to $REDIS_SIZES_KEY"
   - for example in ${INK_EXAMPLES_PATH}/*/; do
-      echo "cargo contract build --manifest-path=${example}/Cargo.toml" >> /tmp/cmds;
+      echo "./.scripts/build-contract.sh ${example} | redis-cli -u ${GITLAB_REDIS_URI} -x rpush ${REDIS_SIZES_KEY}" >> /tmp/cmds;
     done
   - for contract in ${DELEGATOR_SUBCONTRACTS}; do
-      echo "cargo contract build --manifest-path=${INK_EXAMPLES_PATH}/delegator/${contract}/Cargo.toml" >> /tmp/cmds;
+      echo "./.scripts/build-contract.sh ${INK_EXAMPLES_PATH}/delegator/${contract} | redis-cli -u ${GITLAB_REDIS_URI} -x rpush ${REDIS_SIZES_KEY}" >> /tmp/cmds;
```

Comment:
Can we break this into multiple lines? It basically goes from one side of my laptop screen to the next, lol